### PR TITLE
164

### PR DESCRIPTION
**Pull request by dimarazumov**
Changed TestJobDefinition and TestJobResult following the requirenments.
Added custom serializer/deserializer for LocalDateTime fields in TestJobResult.

### CODE REVIEW

File: foundations/src/main/java/io/yields/math/framework/definition/json/JsonDateTimeDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package io.yields.math.framework.definition.json;
```
Comment (rombautt) [inline foundations/src/main/java/io/yields/math/framework/definition/json/JsonDateTimeDeserializer.java]: could you check the existing dependency on 

<dependency>
            <groupId>com.fasterxml.jackson.datatype</groupId>
            <artifactId>jackson-datatype-jsr310</artifactId>
            <version>2.4.0</version>
        </dependency>

I think this provides these already.

Comment (dimarazumov) [inline foundations/src/main/java/io/yields/math/framework/definition/json/JsonDateTimeDeserializer.java]: I've checked before the implementation actually.
Default serializer/deserializer from jsr310 gives us such JSON object: [2015,8,28,10,20,31,9000000]
I suppose it's better to have standart ISO datetime format like I've implement "2015-08-28T10:20:31"

Comment (jgheerardyn) [inline foundations/src/main/java/io/yields/math/framework/definition/json/JsonDateTimeDeserializer.java]: agree

File: foundations/src/main/java/io/yields/math/framework/definition/TestJobDefinition.java [hunk 2]
```diff
@@ -38,12 +41,26 @@ public TestJobDefinition(List<String> testDefinitionNames, String author, TestJo
         this.project = project;
```
Comment (jgheerardyn) [inline foundations/src/main/java/io/yields/math/framework/definition/TestJobDefinition.java]: Now it is possible to instantiate a TestJobDefinition with UUID being null (either by calling the original constructor on line 37) of by setting uuid field in the new constructor to null. I don't think we should allow this.

File: foundations/src/main/java/io/yields/math/framework/definition/TestJobResult.java [hunk 3]
```diff
@@ -16,43 +16,71 @@
...
+    private String application;
     private String testDefinitionName;
     private LocalDateTime executionTimeStart;
     private LocalDateTime executionTimeEnd;
     private TestJobResultType resultType;
     private String errorMessage;
 
-    public static TestJobResult success(String testDefinitionName, LocalDateTime executionTimeStart, LocalDateTime executionTimeEnd) {
-        return new TestJobResult(testDefinitionName, executionTimeStart, executionTimeEnd, TestJobResultType.SUCCEEDED, null);
+    public static TestJobResult success(String uuid, String application, String testDefinitionName, LocalDateTime executionTimeStart,
```
Comment (jgheerardyn) [inline foundations/src/main/java/io/yields/math/framework/definition/TestJobResult.java]: same comment - in my opinion, uuid cannot be null

**Comment by jgheerardyn** [comment]
@dimarazumov last comment, for next time. If a branch in the yields project corresponds to an issue in Yields-finance, please append branch name with **finance_**, thanks

**Pull request merged**